core/src/main/java/org/frankframework/logging/FrankPropertyLookupProvider.java

```diff
@@ -210,8 +218,89 @@ private static void setLevel(Properties properties) {
 			if("ACC".equalsIgnoreCase(stage) || "PRD".equalsIgnoreCase(stage)) {
 				logLevel = "WARN";
 			}
-			properties.setProperty("log.level", logLevel);
-			System.setProperty("log.level", logLevel);
+			properties.setProperty(LOG_LEVEL_KEY, logLevel);
+			System.setProperty(LOG_LEVEL_KEY, logLevel);
+		}
+	}
+
+	/**
+	 * Checks if log.dir property exists.
+	 * Sets it with findLogDir function.
+	 * If it exists, expand the value and set it again.
+	 */
+	private static void setLogDir(Properties properties) {
+		String originalLogDir = properties.getProperty(LOG_DIR_KEY);
+		File logDir;
+		if (originalLogDir == null) {
+			logDir = findLogDir();
+			if (logDir != null) {
+				LOGGER.info(LOOKUP, "did not find system property [log.dir] found suitable path ["+logDir.getPath()+"]");
+			} else {
+				LOGGER.warn(LOOKUP, "did not find system property [log.dir] and unable to locate it automatically");
+			}
+		} else {
+			if(StringResolver.needsResolution(originalLogDir)) {
+				originalLogDir = StringResolver.substVars(originalLogDir, properties);
+				LOGGER.info(LOOKUP, "found system property [log.dir] which required property expansion to suitable path ["+originalLogDir+"]");
+			}
+			logDir = new File(originalLogDir);
+		}
+
+		if (logDir != null) {
+			// Whether it was previously set or not, overwrite it with the fully-expanded value.
+			String expanded = fixLogDirectorySlashesAndExpand(logDir.getPath());
+
+			System.setProperty(LOG_DIR_KEY, expanded);
+			properties.setProperty(LOG_DIR_KEY, expanded);
 		}
 	}
+
+	/**
+	 * Replace backslashes because log.dir is used in log4j2.xml
+	 * on which substVars is done (see below) which will replace
+	 * double backslashes into one backslash and after that the same
+	 * is done by Log4j:
+	 * https://issues.apache.org/bugzilla/show_bug.cgi?id=22894
+	 * */
+	private static String fixLogDirectorySlashesAndExpand(String directory) {
+		return Path.of(directory).toAbsolutePath().toString().replace("\\", "/");
```

Comment:
Waarom eigenlijk worden backslashes omgezet naar forward slashes?

Waarom niet gewoon het pad in platform-native form gebruiken?

Reply:
Geen idee, zo zat het er al in, ik heb de code puur gecopy-paste.